crossdc-consumer/src/main/java/org/apache/solr/crossdc/consumer/KafkaCrossDcConsumer.java

```diff
@@ -220,14 +285,97 @@ boolean pollAndProcessRequests() {
     return true;
   }
 
+  public void sendBatch(UpdateRequest solrReqBatch, ConsumerRecord<String,MirroredSolrRequest> lastRecord, WorkUnit workUnit) {
+    UpdateRequest finalSolrReqBatch = solrReqBatch;
+    Future<?> future = executor.submit(() -> {
+      IQueueHandler.Result<MirroredSolrRequest> result = messageProcessor.handleItem(new MirroredSolrRequest(finalSolrReqBatch));
+      try {
+        processResult(lastRecord, result);
+      } catch (MirroringException e) {
+        // We don't really know what to do here
+        log.error("Mirroring exception occurred while resubmitting to Kafka. We are going to stop the consumer thread now.", e);
```

Comment:
So this is a real error because kafka thinks the updates are processed, but there were Solr errors and we can't resubmit?

Reply:
This is from the original code we contributed, but it should be that we failed sending to solr and we failed re queing to Kafka, so it seems everything is down.